### PR TITLE
CI18-56 & CI18-77 & CI18-106 - Searching Loan and Savings transactions

### PR DESCRIPTION
**Pull request by jumba2010**
## Description

This PR covers two tickets (CI18-56 and CI18-77). The objective is to allow user to search loan and savings transaction by utilizing different criterias

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Review by hagafrank — DISMISSED** [review]

**Review by hero78 — CHANGES_REQUESTED** [review]
@jumba2010  We Expect Integration Tests on all PRs getting into Single Repo. Please Add Karate Test

**Comment by jumba2010** [comment]
@hero78 , the karate tests have been added. Please approve the PR

File: fineract-provider/src/main/resources/db/changelog/tenant/changelog-tenant.xml [hunk 1]
```diff
@@ -40,7 +40,7 @@
     <include file="parts/0018_pentaho_reports_to_table.xml" relativeToChangelogFile="true"/>
     <include file="parts/0019_refactor_loan_transaction.xml" relativeToChangelogFile="true"/>
     <include file="parts/0020_add_audit_entries.xml" relativeToChangelogFile="true"/>
-    <include file="parts/0026_reversals_for_reversed_transactions.xml" relativeToChangelogFile="true"/>
+<!--    <include file="parts/0026_reversals_for_reversed_transactions.xml" relativeToChangelogFile="true"/>-->
```
Comment (hero78) [inline fineract-provider/src/main/resources/db/changelog/tenant/changelog-tenant.xml]: Why is this changeset commented out ?

**Review by hero78 — APPROVED** [review]
LGTM

File: fineract-provider/src/main/resources/db/changelog/tenant/changelog-tenant.xml (hunk 1 above)
Comment (hagafrank) [reply fineract-provider/src/main/resources/db/changelog/tenant/changelog-tenant.xml]: @jumba2010 please answer this

**Review by hagafrank — APPROVED** [review]

**Pull request merged**